src/script/conversation/ConversationRepository.test.ts

```diff
@@ -1262,5 +1265,340 @@ describe('ConversationRepository', () => {
+              self: {},
+            },
+            name: 'conv1',
+            protocol: 'proteus',
+            qualified_id: {
+              domain: 'staging.zinfra.io',
+              id: '05d0f240-bfe9-40d7-b6cb-602dac89fa1b',
+            },
+            receipt_mode: 1,
+            team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
+            type: 0,
+          },
+        ],
+        failed: [
+          {
+            members: {
+              others: [] as QualifiedId[],
+              self: {},
+            },
+            name: 'conv2',
+            protocol: 'proteus',
+            qualified_id: {
+              domain: 'staging.zinfra.io',
+              id: '05d0f240-bfe9-40d7-1234-602dac89fa1b',
+            },
+            receipt_mode: 1,
+            team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
+            type: 0,
+          },
+          {
+            members: {
+              others: [] as QualifiedId[],
+              self: {},
+            },
+            name: 'conv3',
+            protocol: 'proteus',
+            qualified_id: {
+              domain: 'staging.zinfra.io',
+              id: '05d0f240-bfe9-40d7-3212-602dac89fa1b',
+            },
+            receipt_mode: 1,
+            team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
+            type: 0,
+          },
+        ],
+      };
+
+      jest
+        .spyOn(conversationService, 'getAllConversations')
+        .mockResolvedValue(remoteConversations as unknown as RemoteConversations);
+
+      await conversationRepository.loadConversations();
+
+      expect(conversationState.missingConversations).toHaveLength(remoteConversations.failed.length);
+    });
+  });
+  describe('loadMissingConversations', () => {
+    beforeEach(() => {
+      testFactory.conversation_repository!['conversationState'].conversations.removeAll();
+    });
+
+    it('make sure missing conversations are properly updated', async () => {
+      const conversationRepository = testFactory.conversation_repository!;
+      const conversationService = conversationRepository['conversationService'];
+      const conversationState = conversationRepository['conversationState'];
+
+      const missingConversations = [
+        {
+          domain: 'staging.zinfra.io',
+          id: '05d0f240-bfe9-40d7-b6cb-602dac89fa1b',
+        },
+        {
+          domain: 'staging.zinfra.io',
+          id: '05d0f240-bfe9-40d7-1234-602dac89fa1b',
+        },
+      ];
+
+      const remoteConversations = {
+        found: [
+          {
+            members: {
+              others: [] as QualifiedId[],
+              self: {},
+            },
+            name: 'conv1',
+            protocol: 'proteus',
+            qualified_id: {
+              domain: 'staging.zinfra.io',
+              id: '05d0f240-bfe9-40d7-b6cb-602dac89fa1b',
+            },
+            receipt_mode: 1,
+            team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
+            type: 0,
+          },
```

Comment:
There are a lot of variations of those lines. You could write a generator function:

```js
function generateConversation(name: string) {
   return {
            members: {
              others: [] as QualifiedId[],
              self: {},
            },
            name,
            protocol: 'proteus',
            qualified_id: {
              domain: 'staging.zinfra.io',
              id: '05d0f240-bfe9-40d7-b6cb-602dac89fa1b',
            },
            receipt_mode: 1,
            team: 'b0dcee1f-c64e-4d40-8b50-5baf932906b8',
            type: 0,
          },
}
```

(eventually adding more parameters if you need)